### PR TITLE
Bugfix: Remove illegal chars

### PR DESCRIPTION
**Pull request by joeflack4**
## Updates
Remove illegal unicode characters found.

## Related
- https://github.com/monarch-initiative/mondo-ingest/pull/412
- https://github.com/monarch-initiative/mondo-ingest/pull/434

---

@twhetzel Thanks for helping narrow the problem!

### CODE REVIEW

File: icd11_foundation_ingest/clean_illegal_chars.py [hunk 1]
```diff
@@ -0,0 +1,23 @@
+"""Clean illegal chars"""
+from argparse import ArgumentParser
+
+
+def clean(inpath: str, outpath: str):
+    """Clean"""
+    with open(inpath, 'rb') as file:
+        content = file.read()
+    cleaned_content = content.replace(b'\x03', b'')
```
Comment (joeflack4) [inline icd11_foundation_ingest/clean_illegal_chars.py]: The fix

File: makefile [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+	mkdir -p $@
+
+tmp/input/:
+	mkdir -p $@
+
+tmp/input/source.owl: | tmp/input/
+	wget ${SOURCE_URL} -O $@
+
+# Requires GitHub CLI: https://cli.github.com/
+deploy-release: | tmp/output/release/
```
Comment (joeflack4) [inline makefile]: Deploy release command using GitHub CLI

File: makefile [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+TODAY ?=$(shell date +%Y-%m-%d)
+VERSION=v$(TODAY)
+SOURCE_URL=https://icd11files.blob.core.windows.net/tmp/whofic-2023-04-08.owl.gz
+
+
+# MAIN COMMANDS / GOALS ------------------------------------------------------------------------------------------------
+all: tmp/output/release/icd11foundation.owl
+
+tmp/output/release/icd11foundation.owl: tmp/input/source.owl | tmp/output/release/
+	 python3 icd11_foundation_ingest/clean_illegal_chars.py -i $< -o $@
```
Comment (joeflack4) [inline makefile]: Goal for fix and release artefact. Also runs when running `make all`.

File: makefile (hunk 2 above)
Comment (joeflack4) [reply makefile]: Release is up: https://github.com/monarch-initiative/icd11/releases
Updating `mondo-ingest` now to use.

**Comment by twhetzel** [comment]
@joeflack4 while I haven't tried to run this yet, looks like this solves the issues. While I'm not opposed to the python script, did you also have a chance to try adding a unix call, e.g. `tr '\003' ' ' < input_file > output_file`, into the workflow which is a bit more straightforward

File: makefile (hunk 3 above)
Comment (joeflack4) [reply makefile]: [Trish wrote](https://github.com/monarch-initiative/icd11/pull/2#issuecomment-1949112751):
> @joeflack4 while I haven't tried to run this yet, looks like this solves the issues. While I'm not opposed to the python script, did you also have a chance to try adding a unix call, e.g. `tr '\003' ' ' < input_file > output_file`, into the workflow which is a bit more straightforward

That is a good idea; I shoulda reached for a unix command, I've never actually seen `tr` before. Anyway as we spoke about on the call, there are more such illegal chars, so my first priority is to reach for a unix solution that removes all such illegal unicode chars.

Comment (twhetzel) [reply makefile]: Here's a more general Unix command (from ChatGPT) that works by deleting all characters that are not in the ASCII range. When `-cd` are used together, `-c` complements the set, and `-d` deletes the characters that match the resulting set. This combination effectively removes all characters from the input stream that are not in the specified ASCII range, effectively removing Unicode characters.

`tr -cd '\11\12\15\40-\176' < yourfile.txt > yourfile-without-unicode.txt`

I ran this on "component-download-icd11foundation.owl.owl" and the resulting file could be opened by Protege and in a quick look over the labels they seem fine.

Comment (joeflack4) [reply makefile]: @twhetzel That's amazing! ChatGPT is a good old chap. Thanks for the help. I'll use this solution!

Comment (joeflack4) [reply makefile]: Worked like a charm! `mondo-ingest` has no trouble parsing now.

This can be merged unless there are other requests for tweaks.

**Comment by joeflack4** [comment]
This has been ready to merge, and I'm merging it now for ease of making this new PR (#3) to address [equivalence issues](https://github.com/monarch-initiative/mondo-ingest/pull/412#discussion_r1493469285).

**Comment by twhetzel** [comment]
This looks ok.

**Pull request merged**